amazon-kinesis-connector-flink/src/main/java/software/amazon/kinesis/connectors/flink/internals/KinesisDataFetcher.java

```diff
@@ -1222,14 +1224,19 @@ public void onProcessingTime(long timestamp) {
 	 *
 	 * @return a {@link MetricGroup} that can be used to update metric values
 	 */
-	private MetricGroup registerShardMetricGroup(final MetricGroup metricGroup, final KinesisStreamShardState shardState) {
-		return metricGroup
-				.addGroup(
-						KinesisConsumerMetricConstants.STREAM_METRICS_GROUP,
-						shardState.getStreamShardHandle().getStreamName())
-				.addGroup(
-						KinesisConsumerMetricConstants.SHARD_METRICS_GROUP,
-						shardState.getStreamShardHandle().getShard().getShardId());
+	public static GenericMetricGroup registerShardMetricGroup(final MetricGroup metricGroup, final StreamShardHandle shardHandle) {
+		try {
+			return (GenericMetricGroup) metricGroup
```

Comment:
Can you cast to `AbstractMetricGroup` instead?

Reply:
Updated to `AbstractMetricGroup`